src/DIRAC/TransformationSystem/Service/TransformationManagerHandler.py

```diff
@@ -33,6 +34,29 @@ def initializeHandler(cls, serviceInfoDict):
 
         return S_OK()
 
+    def checkPermissions(self, transName):
+        """
+        checks if remote user has permission to access to a given transformation
+        :param str transName: Name of the transformation to check
+        :return: False if user does not have permission or if transformation does not exist
+        """
+        credDict = self.getRemoteCredentials()
+        groupProperties = credDict.get("properties", [])
+        if SecurityProperty.PRODUCTION_MANAGEMENT in groupProperties:
+            return True
+        tfDetails = self.transformationDB.getTransformation(transName)
+        if not tfDetails["OK"]:
+            return False
+        authorGroup = tfDetails["Value"]["AuthorGroup"]
+        authorDN = tfDetails["Value"]["AuthorDN"]
+        if SecurityProperty.PRODUCTION_SHARING in groupProperties:
+            if authorGroup == credDict["group"]:
+                return True
+        if SecurityProperty.PRODUCTION_USER in groupProperties:
+            if authorDN == credDict["DN"]:
```

Comment:
```suggestion
            if authorDN == credDict["DN"]:
```
This needs to change to author_username for 8.1 (?)

Reply:
Yes. And in general no new dependencies on DNs should be introduced here, so please refactor using `username` instead of DN.

Reply:
@fstagni @andresailer It is my understanding that we cannot use username in 8.0, so we would have to do two different version of the code, one for 8.0 and one for 8.1. I don't want this to only go in 8.1, at the moment the security model is too loose for comfort for Simon to use. We would just tag this request as "sweep-ignore", so it should not be carried forward and make a different request for integration ?
@andresailer Shouldn't the code be already updated to use username in integration ?

Reply:
Actually it seems the TSManager was not yet changed to username instead of authorDN.

So I think it is fine to keep using DN here, because otherwise you would need to get the username from the authorDN, instead of comparing credDict DN with authorDN. (It is just one place anyway where this happens, so easy to adapt later)

And if necessary the swept PR could be changed, but as the TSManager wasn't yet changed...

Reply:
> @fstagni @andresailer It is my understanding that we cannot use username in 8.0, so we would have to do two different version of the code, one for 8.0 and one for 8.1. I don't want this to only go in 8.1, at the moment the security model is too loose for comfort for Simon to use. We would just tag this request as "sweep-ignore", so it should not be carried forward and make a different request for integration ? @andresailer Shouldn't the code be already updated to use username in integration ?

`self.getRemoteCredentials()['username']` would work, no?

Reply:
But then you have to convert either the authorDN to username, or the username to DN, which seems really superfluous, if you can get the DN from the credDict

Reply:
I don't see a perfect solution here. @marianne013 Then don't sweep this PR but create a new one.

Reply:
OK, will do (Simon and me are currently in the IAM workshop, so it won't happen today or tomorrow).
@andresailer Could you update the existing code in integration to username, so we are sure it actually works before I go and make any modifications ? Then we'll fix the issues raised in this request, push for 8.0 only, and once that is done, we dump our improvements on Andre's updates in integration.

Reply:
I noticed that unlike other systems which seem to have an update path from DN to username, this doesn't seem to exist for Transformations (yet?): 
https://github.com/DIRACGrid/DIRAC/wiki/DIRAC-8.1
That seems above my paygrade.

Reply:
I'll take care of that.

Reply:
Done in #7124  -- have a look.